### PR TITLE
[KOB-48309][KOB-43713] Enhance Appium Script on scrollable (web)

### PR DESCRIPTION
**Pull request by chuong777**
## Types of changes

_What types of changes does your code introduce to this project? Put an `x` in the boxes that apply._

- [x] Bugfix.
- [ ] New feature.
- [ ] Documentation update.

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the pull request. If you are unsure about any of them, do not hesitate to ask. We are here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) documentation.
- [x] I have rebased, and my remote branch is now up-to-date with the `master` branch of the upstream.
- [x] Lint and unit tests pass locally with my changes.
- [x] I have added tests that prove my changes are effective.
- [x] I have added the necessary documentation (if appropriate).

### CODE REVIEW

File: src/templates/csharp/TestBase.cs [hunk 1]
```diff
@@ -443,6 +461,73 @@ public List<AppiumWebElement> FindElementsBy(params By[] locators)
...
+                (attempt) =>
+                {
+                    var foundElement = FindElementBy(locators);
+                    var rect = foundElement.Rect;
+                    if (rect.X < 0 || rect.Y < 0)
+                    {
+                        throw new Exception("Element is found but is not visible");
+                    }
+
+                    return foundElement;
```
Comment (TranKienCuong) [inline src/templates/csharp/TestBase.cs:485]: Xium has the endpoint `/session/:sessionId/element/:elementId/location_in_view` that will automatically scroll to the element to make this visible and then return the element position on the screen. Could you give it a try if needed?

File: src/templates/csharp/TestBase.cs [hunk 2]
```diff
@@ -105,133 +105,149 @@ public void SwitchToNativeContext()
...
+                    catch (Exception ex)
+                    {
+                        Log($"Bad context {context}, error \"{ex.Message}\", skipping...");
+                        continue;
+                    }
 
-                        HtmlDocument htmlDoc = LoadXMLFromString(source);
-                        HtmlNode bodyElements = htmlDoc.DocumentNode.SelectSingleNode("/html/body");
-                        if (bodyElements == null) continue;
+                    if (source == null) continue;
```
Comment (TranKienCuong) [inline src/templates/csharp/TestBase.cs:148]: For your reference, we can add one more valuable condition here to get rid of invalid/hidden web contexts, just execute the script `document.hidden` to check if it's visible or not.
Ref:
- https://github.com/kobiton/ita-core-util/blob/main/src/screen-info-builder/index.ts#L448
- https://github.com/kobiton/static-resources-manager/blob/master/src/bin/ksl-files/kobiton-inventory.js#L62

File: src/templates/csharp/TestBase.cs (hunk 1 above)
Comment (chuong777) [reply src/templates/csharp/TestBase.cs]: The appium script run on both Xium and Appium, so we only use endpoint that both server support

File: src/templates/csharp/TestBase.cs (hunk 2 above)
Comment (chuong777) [reply src/templates/csharp/TestBase.cs:154]: I will apply this logic on next PR. Thank you!

**Review by d4rkwinz — APPROVED** [review]

**Pull request merged**